zkvm/src/predicate.rs

```diff
@@ -27,19 +29,45 @@ pub enum Predicate {
     /// constructing aggregated signature.
     Key(VerificationKey),
 
-    /// Representation of a predicate as commitment to a program and blinding factor.
-    Program(Program, Vec<u8>),
-
-    /// Disjunction of n predicates.
-    Or(PredicateDisjunction),
+    /// Taproot Merkle tree.
+    Tree(PredicateTree),
 }
 
+/// Represents a ZkVM predicate tree.
 #[derive(Clone, Debug)]
-pub struct PredicateDisjunction {
-    preds: Vec<Predicate>,
-    // We precompute the disjunction predicate when composing it via `Predicate::or()`,
+pub struct PredicateTree {
+    /// Vector of the programs and blinding factors, stored as Merkelized predicate leafs.
+    pub leaves: Vec<PredicateLeaf>,
```

Comment:
keeping these fields public allows holder of mut-borrow to modify the fields directly and getting other private fields out of sync. Let's make them private and expose necessary getter methods if needed?